test/uk/gov/hmrc/tradergoodsprofiles/controllers/WithdrawAdviceControllerSpec.scala

```diff
@@ -0,0 +1,111 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofiles.controllers
+
+import org.mockito.ArgumentMatchersSugar.{any, eqTo}
+import org.mockito.MockitoSugar.{reset, verify, when}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatestplus.mockito.MockitoSugar.mock
+import org.scalatestplus.play.PlaySpec
+import play.api.http.Status.{INTERNAL_SERVER_ERROR, NO_CONTENT, OK}
+import play.api.libs.json.Json
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{contentAsJson, defaultAwaitTimeout, status, stubControllerComponents}
+import uk.gov.hmrc.tradergoodsprofiles.connectors.WithdrawAdviceRouterConnector
+import uk.gov.hmrc.tradergoodsprofiles.controllers.support.AuthTestSupport
+import uk.gov.hmrc.tradergoodsprofiles.controllers.support.FakeAuth.FakeSuccessAuthAction
+import uk.gov.hmrc.tradergoodsprofiles.models.errors.{ErrorResponse, ServiceError}
+import uk.gov.hmrc.tradergoodsprofiles.services.UuidService
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+class WithdrawAdviceControllerSpec extends PlaySpec with AuthTestSupport with BeforeAndAfterEach {
+
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  private val request = FakeRequest()
+    .withHeaders(
+      "Accept"       -> "application/vnd.hmrc.1.0+json",
+      "Content-Type" -> "application/json",
+      "X-Client-ID"  -> "some client ID"
+    )
+
+  val requestHeaders: Seq[(String, String)] = Seq(
+    "Accept"       -> "application/vnd.hmrc.1.0+json",
+    "Content-Type" -> "application/json"
+  )
+  private val recordId                      = UUID.randomUUID().toString
+  private val correlationId                 = "d677693e-9981-4ee3-8574-654981ebe606"
+  private val withdrawReason                = "today"
+  private val uuidService                   = mock[UuidService]
+  private val connector                     = mock[WithdrawAdviceRouterConnector]
+  private val sut                           = new WithdrawAdviceController(
+    new FakeSuccessAuthAction(),
+    connector,
+    uuidService,
+    stubControllerComponents()
+  )
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    reset(uuidService, connector)
+    when(uuidService.uuid).thenReturn(correlationId)
+    when(connector.withdrawAdvice(any, any, any)(any))
+      .thenReturn(Future.successful(Right(OK)))
+  }
+
+  "withdrawAdvice" should {
+    "return 204" in {
```

Comment:
You can remove this test, the httpStatus is being tested again in the test below